chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/FilterDAOImpl.java

```diff
@@ -0,0 +1,158 @@
+package gov.healthit.chpl.dao.impl;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.stereotype.Repository;
+
+import gov.healthit.chpl.auth.Util;
+import gov.healthit.chpl.auth.entity.UserEntity;
+import gov.healthit.chpl.dao.FilterDAO;
+import gov.healthit.chpl.dto.FilterDTO;
+import gov.healthit.chpl.dto.FilterTypeDTO;
+import gov.healthit.chpl.entity.FilterEntity;
+import gov.healthit.chpl.entity.FilterTypeEntity;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+
+@Repository("filterDAO")
+public class FilterDAOImpl extends BaseDAOImpl implements FilterDAO {
+
+    @Override
+    public FilterDTO update(final FilterDTO dto) throws EntityRetrievalException {
+        FilterEntity entity = new FilterEntity();
```

Comment:
What is this updating? I thought for an update method we normally would get the FilterEntity by ID and then fill in the rest of the values as you've done but this is creating a new FilterEntity and never setting the ID field. Does it actually end up inserting a new row or does it update an existing one?

Reply:
The update method is there just for completeness of the DAO.  It is not being called, therefore it was nor really tested.  Method has been updated.